src/back/bootstrap/index.js

```diff
@@ -9,10 +9,18 @@ const
 
 module.exports = (app) => {
     db(formatMongooseDbUriFromEnv(env));
+    extendRequest();
     preMiddleware(app);
     return app.use(controllers);
 };
 
+function extendRequest() {
+    const authService = require('../controllers/auth/auth.service'),
+        http = require('http');
+    http.IncomingMessage.prototype.getCurrentUser = authService.getCurrentUser;
```

Comment:
I don't like monkey patch prototypes, why did not just create middleware for _strict_ and _optional_ authentication and protect endpoints with them